tokens/batched/tokens.go

```diff
@@ -19,11 +21,39 @@ func UnmarshalBatchedTokenResponses(data []byte) ([][]byte, error) {
 
 	var token_responses [][]byte
 	for !token_responses_string.Empty() {
-		var token_response cryptobyte.String
-		if !token_responses_string.ReadUint16LengthPrefixed(&token_response) {
+		var present uint8
+
+		if !token_responses_string.ReadUint8(&present) {
+			return nil, fmt.Errorf("invalid Token encoding")
+		}
+
+		if present == uint8(TokenStatusAbsent) {
+			token_responses = append(token_responses, []byte{})
+		} else if present == uint8(TokenStatusPresent) {
+			var token_type uint16
+			if !token_responses_string.ReadUint16(&token_type) {
+				return nil, fmt.Errorf("invalid Token encoding")
+			}
+
+			var token_response_length uint16
+			switch token_type {
+			case type1.BasicPrivateTokenType:
+				token_response_length = 145 // Ne + 2*Ns
```

Comment:
it would be good to define constants `Ne` `Ns`

Reply:
done, as well as opened #57